gcloud/datastore/__init__.py

```diff
@@ -170,4 +182,7 @@ def allocate_ids(incomplete_key, num_ids):
     :rtype: list of :class:`gcloud.datastore.key.Key`
     :returns: The (complete) keys allocated with `incomplete_key` as root.
     """
-    return _require_dataset().allocate_ids(incomplete_key, num_ids)
+    dataset = _require_dataset()
+    connection = _require_connection()
+    return connection_module.allocate_ids(incomplete_key, num_ids,
```

Comment:
That would be fine with me (will they take `connection` and `dataset_id` as optional args, falling back to the implicit ones?)

They will have to muck with protobufs, which might have been why we thought to land them in `connection`.